Util/src/main/java/io/deephaven/util/type/ArrayTypeUtils.java

```diff
@@ -426,70 +302,64 @@ public static Object[] getBoxedArray(Object value) {
+        } else if (actualValue instanceof char[] && expectedValue instanceof char[]) {
             return Arrays.equals((char[]) actualValue, (char[]) expectedValue);
-        } else if (double.class.isAssignableFrom(ct)) {
+        } else if (actualValue instanceof double[] && expectedValue instanceof double[]) {
             return Arrays.equals((double[]) actualValue, (double[]) expectedValue);
-        } else if (float.class.isAssignableFrom(ct)) {
+        } else if (actualValue instanceof float[] && expectedValue instanceof float[]) {
             return Arrays.equals((float[]) actualValue, (float[]) expectedValue);
-        } else if (int.class.isAssignableFrom(ct)) {
+        } else if (actualValue instanceof int[] && expectedValue instanceof int[]) {
             return Arrays.equals((int[]) actualValue, (int[]) expectedValue);
-        } else if (long.class.isAssignableFrom(ct)) {
+        } else if (actualValue instanceof long[] && expectedValue instanceof long[]) {
             return Arrays.equals((long[]) actualValue, (long[]) expectedValue);
-        } else if (short.class.isAssignableFrom(ct)) {
+        } else if (actualValue instanceof short[] && expectedValue instanceof short[]) {
             return Arrays.equals((short[]) actualValue, (short[]) expectedValue);
+        } else {
+            return Arrays.equals((Object[]) actualValue, (Object[]) expectedValue);
         }
-        return false;
     }
 
     public static String toString(Object actualValue) {
-        final Class<?> ct = actualValue.getClass().getComponentType();
-        if (Object.class.isAssignableFrom(ct)) {
-            return Arrays.toString((Object[]) actualValue);
-        } else if (byte.class.isAssignableFrom(ct)) {
+        if (actualValue instanceof byte[]) {
             return Arrays.toString((byte[]) actualValue);
-        } else if (char.class.isAssignableFrom(ct)) {
+        } else if (actualValue instanceof char[]) {
             return Arrays.toString((char[]) actualValue);
-        } else if (double.class.isAssignableFrom(ct)) {
+        } else if (actualValue instanceof double[]) {
             return Arrays.toString((double[]) actualValue);
-        } else if (float.class.isAssignableFrom(ct)) {
+        } else if (actualValue instanceof float[]) {
             return Arrays.toString((float[]) actualValue);
-        } else if (int.class.isAssignableFrom(ct)) {
+        } else if (actualValue instanceof int[]) {
             return Arrays.toString((int[]) actualValue);
-        } else if (long.class.isAssignableFrom(ct)) {
+        } else if (actualValue instanceof long[]) {
             return Arrays.toString((long[]) actualValue);
-        } else if (short.class.isAssignableFrom(ct)) {
+        } else if (actualValue instanceof short[]) {
             return Arrays.toString((short[]) actualValue);
         }
-        return null;
+        return Arrays.toString((Object[]) actualValue);
```

Comment:
Ditto; I think we want to check `instanceof Object[]`.

Reply:
Yep, can change, but callers _appear_ to all require that these are arrays anyway

Reply:
Other examples where this behavior (casting Object to Object[]) is expected in this class:

- getBoxedArray
- getArrayAccessor

Reply:
Same as above. CCE might be just as good as UOE.